src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -388,37 +388,45 @@
   Respects *nested-field-column-max-row-length*."
   [v path]
   (let [p (json-parser v)]
-    (loop [path (or path [])
-           key  nil
-           res  (transient {})]
-      (let [token     (.nextToken p)
-            next-path (cond-> path key (conj key))]
+    (loop [path      (or path [])
```

Comment:
The fix here is reverese https://github.com/metabase/metabase/pull/43812/files, then add a check to make sure we return nil if the value is not an object.